### PR TITLE
etcd-tester: add '-failpoints' to configure gofail

### PR DESCRIPTION
**Pull request by gyuho**
Sleep 5s

Or just 1-sec?

### CODE REVIEW

**Comment by heyitsanthony** [comment]
does this slow down the rounds much more than panic failpoints?

**Comment by gyuho** [comment]
@heyitsanthony Yes. The total rounds used to take ~15min, now it takes +20min. Should we make the duration configurable?

**Comment by heyitsanthony** [comment]
@gyuho not super concerned about total round time since more test cases => longer rounds. More interested whether a longer sleep duration will cause the sleep failpoint tests to take longer...

Having some config option could be good since there's a lot of failpoint actions which may be useful for uncovering bugs (e.g., crash immediately after sleeping with `sleep(1000)->panic`, randomize sleeping with `10%sleep(1000)`). I'm thinking the tester could take a list of failpoint actions and construct test cases from them instead of hardcoding it all

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Now it's configurable with `-failpoints` flag.

Thanks.

File: tools/functional-tester/etcd-tester/main.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func main() {
 	stresserType := flag.String("stresser", "keys,lease", "comma separated list of stressers (keys, lease, v2keys, nop, election-runner, watch-runner, lock-racer-runner, lease-runner).")
 	etcdRunnerPath := flag.String("etcd-runner", "", "specify a path of etcd runner binary")
 	failureTypes := flag.String("failures", "default,failpoints", "specify failures (concat of \"default\" and \"failpoints\").")
+	failpoints := flag.String("failpoints", `panic("etcd-tester")`, `define failpoints (e.g. concat of panic("etcd-tester") and 1*sleep(1000)->panic("etcd-tester")`)
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-tester/main.go]: `comma separated list of failpoint terms to inject (e.g. 'panic("etcd-tester"),1*sleep(1000)')`

File: tools/functional-tester/etcd-tester/main.go [hunk 2]
```diff
@@ -177,7 +178,7 @@ func portsFromArg(arg string, n, defaultPort int) []int {
 	return ret
 }
 
-func makeFailures(types string, c *cluster) []failure {
+func makeFailures(types, failpoints string, c *cluster) []failure {
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-tester/main.go]: Make everything explicit slices from the start?
```go
func makeFailures(types, failpoints []string, c *cluster) []failure
```

File: tools/functional-tester/etcd-tester/main.go [hunk 3]
```diff
@@ -82,7 +83,7 @@ func main() {
 	var failures []failure
 
 	if failureTypes != nil && *failureTypes != "" {
-		failures = makeFailures(*failureTypes, c)
+		failures = makeFailures(*failureTypes, *failpoints, c)
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-tester/main.go]:
```go
types, failpoints := strings.Split(*failureTypes, ","), strings.Split(*failpoints, ",")
failures = makeFailures(types, failpoints, c)
```
?

**Comment by gyuho** [comment]
@heyitsanthony PTAL. Thanks!

**Comment by heyitsanthony** [comment]
lgtm thanks

**Pull request merged**